android/src/main/java/com/expensify/livemarkdown/MarkdownFormatter.java

```diff
@@ -24,8 +27,19 @@ public void format(SpannableStringBuilder ssb, List<MarkdownRange> markdownRange
     try {
       Systrace.beginSection(0, "format");
       Objects.requireNonNull(markdownStyle, "mMarkdownStyle is null");
+
+      String text = ssb.toString();
+      if (text.equals(mPrevText) && markdownRanges == mPrevMarkdownRanges && markdownStyle == mPrevMarkdownStyle) {
+        // Use shallow comparison of markdown ranges and markdown style
+        // to optimistically skip removing and applying the same spans
+        return;
+      }
```

Comment:
Isn't there anything that could alter the styles in between format passes? I'm not sure how it will behave with the commit hook when the text doesn't change but the node is cloned.